blog/2023/2023-04-17-writing-your-first-pull-request-tips-best-practices-and-ai-powered-tools-for-success.md

```diff
@@ -0,0 +1,65 @@
+---
+title: "Writing Your First Pull Request: Tips, Best Practices, and AI-Powered Tools for Success"
+tags: []
+authors: BekahHW
+slug: writing-your-first-pull-request-tips-best-practices-and-ai-powered-tools-for-success
+description: "If you're new to contributing to open-source projects, submitting your first pull request can be intimidating. But don't worry, we've got you covered! Check out these tips for successfully submitting your first pull request and making a great first impression as a contributor. "
+image: https://media.dev.to/cdn-cgi/image/width=1000,height=420,fit=cover,gravity=auto,format=auto/https%3A%2F%2Fdev-to-uploads.s3.amazonaws.com%2Fuploads%2Farticles%2F4xtz1foaoswl7f55hzgk.png
+---
+
+I don’t remember writing my first Pull Request (PR), and to be honest, I actually looked to see if I could find it for this post, but I know that I was really nervous because I wasn’t sure if I was doing it “right.” I think I actually spent more time worrying about it than I did actually writing the thing. The good news is that there isn’t really a right way to write a PR, but there are some best practices that can guide your writing and help the maintainer to be able to understand the changes. 
+
+<!-- truncate -->
+
+## Review the Contribution Guidelines
+Maintainers want to help you write your PR; that’s why they often outline contributing directions in their repositories.
+
+![Jerry McGuire “Help me, help you gif”](https://media.giphy.com/media/uRb2p09vY8lEs/giphy.gif)
+
+This file—often called CONTRIBUTING.md—gives guidelines for submitting your PR, including information like whether or not you need to write tests, the communication process, coding style, and how to submit your pull request. Check out [OpenSauced’s Contributing Guide](https://docs.opensauced.pizza/contributing/introduction-to-contributing/), as an example. 
+
+## Tips for Your First PR
+Although every repository may have unique guidelines, there are general guidelines for all PRs. Here are some tips to keep in mind:
+
+### Create a Clear Title
+The title should give an indication of the issue, the changes made, and the type of PR–is this a bug fix, content or documentation change, new feature, or something else? 
+
+![OpenSauced title examples](https://dev-to-uploads.s3.amazonaws.com/uploads/articles/yv10emgx8a7grxyovbew.png)
```

Comment:
Are we OK with the images coming from dev.to or do we want them on our own site? I'd suggest longterm getting them in our site so we own all of it and in the event that DEV ever goes away (hopefully not), we'll have all our assets.

Reply:
The goal is to just get this in right now and replace them with our own as time permits.